From 2820ca36ab21c52341cdbde477756f960eaeb68b Mon Sep 17 00:00:00 2001
From: Cheng C Yang <cheng.c.yang@intel.com>
Date: Wed, 13 Mar 2019 11:19:07 +0800
Subject: [PATCH] Create dbus interface for SOL commands

Create dbus properties for Set/Get SOL config parameter command.
Some platforms need to call Set/Get SOL config parameter command
through KCS, and since sol manager in net-ipmid cannot be accessed
by commands in host-ipmid, need to create a dbus interface in
phospher-settings to transfer properties from host-ipmid to
net-ipmid.

TestedBy:
With the related code change in net-ipmid, busctl introspect
xyz.openbmc_project.Ipmi.SOL /xyz/openbmc_project/SOL
can show all the properties needed.

Signed-off-by: Cheng C Yang <cheng.c.yang@intel.com>
---
 xyz/openbmc_project/Ipmi/SOL.interface.yaml | 44 ++++++++++++++++++++++++++
 1 file changed, 44 insertions(+)
 create mode 100644 xyz/openbmc_project/Ipmi/SOL.interface.yaml

diff --git a/xyz/openbmc_project/Ipmi/SOL.interface.yaml b/xyz/openbmc_project/Ipmi/SOL.interface.yaml
new file mode 100644
index 0000000..94db59f
--- /dev/null
+++ b/xyz/openbmc_project/Ipmi/SOL.interface.yaml
@@ -0,0 +1,44 @@
+description: >
+    SOL properties use for Get/Set SOL config parameter command in host-ipmid
+    sending config to SOL process in net-ipmid.
+    Since some platforms need to access Get/Set SOL config parameter command
+    through KCS, and current sol manager is implemented in net-ipmid and
+    cannot be accessed by host-ipmid, add a dbus interface for host-ipmid
+    command to transfer properties to net-ipmid.
+    This interface will be implemented in phosphor-settings.
+properties:
+    - name: Progress
+      type: byte
+      description: >
+        Set In Progress property, indicate when any parameters are being
+        updated.
+    - name: Enable
+      type: boolean
+      description: >
+        SOL Enable property, this controls whether the SOL payload type
+        can be activated.
+    - name: Authentication
+      type: byte
+      description: >
+        If SOL enable Force Payload Encryption and Authenticaton.
+        And the minimun operating privilege level SOL required.
+    - name: AccumulateIntervalMS
+      type: byte
+      description: >
+        Character Accumulate Interval in 5ms increments.
+        BMC will wait this time before transmitting a packet.
+    - name: Threshold
+      type: byte
+      description: >
+        BMC will automatically send an SOL character data packet containing
+        this number of characters as soon as this number of characters
+        (or greater) has been accepted from the baseboard serial controller.
+    - name: RetryCount
+      type: byte
+      description: >
+        Packet will be dropped if no ACK/NACK received by time retries
+        expire.
+    - name: RetryIntervalMS
+      type: byte
+      description: >
+        Retry Interval in 10ms increments.
-- 
2.16.2

